scheduler/scheduler.go

```diff
@@ -132,6 +141,11 @@ func NewScheduler(opts ...Option) *Scheduler {
 	}
 	s.resourceSem = semaphore.NewWeighted(int64(resourceConcurrency))
 
+	// To preserve backwards compatibility, if singleResourceMaxConcurrency is not set, set it to the max concurrency
+	if s.singleResourceMaxConcurrency == 0 {
+		s.singleResourceMaxConcurrency = int64(resourceConcurrency)
+	}
```

Comment:
I think we don't need to preserve backwards-compatibility for behavior that was not good before, let's rather pick a reasonable default